web/module/machines.py

```diff
@@ -120,27 +120,21 @@ def handle_exceptions(request, exception):
     }
 
 
-async def machine_start(request, machine_id):
-    with data.Connection.use() as conn:
-        asyncio.sleep(0.1)
-        machine = data.Machine.get_one_for_update({'_id': machine_id}, conn=conn)
-        new_request = data.Request(type='start', machine=str(machine_id))
-        new_request.save(conn=conn)
-        machine.requests.append(new_request.id)
-        machine.save(conn=conn)
-        data.Action(type='other', request=new_request.id).save(conn=conn)
+@machines.route('/machines/<machine_id>', methods=['PUT'])
```

Comment:
I've done a bit deduplication here. Method is now only one , the only difference inside is request  type (start or stop).

Reply:
good idea